lib/src/request.dart

```diff
@@ -126,7 +126,7 @@ class Request extends Message {
   // TODO(kevmoo) finish documenting the rest of the arguments.
   Request(String method, Uri requestedUri,
       {String protocolVersion,
-      Map<String, String> headers,
+      Map<String, /* String | List<String> */ dynamic> headers,
```

Comment:
Wonder if we should use `Object` here and below, since we don't want to allow `null`, right? (Thinking forward to NNBD)

Reply:
As an empty `<String>[]` can be used to remove an existing header with the `.change()` call, I don't mind not using `null`.

Reply:
Updated, now the code does not accept `null`.

Reply:
Reverted back to accept `null`, but kept `Object`.